File: typescript/palindromic-substrings.ts
function countSubstrings(s: string): number {
  let palindromes = 0;
  for (let i = 0; i < s.length; i++) {
    palindromes += countPalindromes(s, i, i);
    palindromes += countPalindromes(s, i, i + 1);
  }
  return palindromes;
}

function countPalindromes(str: string, center1: number, center2: number): number {
  let palindromes = 0;
  
  let i = center1;
  let j = center2;
  while (i >= 0 && j < str.length && str[i] === str[j]) {
    palindromes++;
    i--;
    j++;
  }

  return palindromes;
}

import { test, expect } from "bun:test";

test("example 1", () => {
  expect(countSubstrings("abc")).toEqual(3);
  expect(countSubstrings("aaa")).toEqual(6);
});

File: python3/palindromic-substrings.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        def count_palindromes(center1: int, center2: int) -> int:
            count = 0
            while 0 <= center1 and center2 < len(s) and s[center1] == s[center2]:
                count += 1
                center1 -= 1
                center2 += 1
            return count
        count = 0
        for i, _ in enumerate(s):
            count += count_palindromes(i, i)
            count += count_palindromes(i, i + 1)
        return count
